test/Dialect/MSFT/dynamic_instance.mlir

```diff
@@ -0,0 +1,71 @@
+// RUN: circt-opt %s -verify-diagnostics | circt-opt -verify-diagnostics
+// RUN: circt-opt %s --msft-lower-instances | FileCheck %s
+
+msft.instance.dynamic [#hw.innerNameRef<@deeper::@branch>, #hw.innerNameRef<@shallow::@leaf>, #hw.innerNameRef<@leaf::@module>] {
```

Comment:
@mikeurbach What do you think of this?

Reply:
I think it looks great. The whole series of patches looks like a really good direction.